src/step-function.ts

```diff
@@ -466,6 +466,428 @@ export namespace $SFN {
       });
     },
   });
+
+  export const partition = makeStepFunctionIntegration<
+    "SFN.Partition",
+    <T>(arr: T[], size: number) => T[][]
+  >("SFN.Partition", {
+    asl(call, context) {
+      const [arr, size] = call.args;
+
+      if (!arr || !size) {
+        throw new SynthError(
+          ErrorCodes.Invalid_Input,
+          "Expected Partition array and size arguments to be provided."
+        );
+      }
+
+      return context.evalContext(
+        call,
+        ({ evalExprToJsonPath, evalExprToJsonPathOrLiteral }) => {
+          const arrayOut = evalExprToJsonPath(arr.expr);
+          const sizeOut = evalExprToJsonPathOrLiteral(size.expr);
+          if (
+            ASLGraph.isLiteralValue(sizeOut) &&
+            typeof sizeOut.value !== "number"
+          ) {
+            throw new SynthError(
+              ErrorCodes.Invalid_Input,
+              "Expected Partition size argument to be a number or reference."
+            );
+          }
+          const temp = context.newHeapVariable();
+          return {
+            Type: "Pass",
+            Parameters: {
+              "out.$": `States.ArrayPartition(${arrayOut.jsonPath}, ${
+                ASLGraph.isJsonPath(sizeOut) ? sizeOut.jsonPath : sizeOut.value
+              })`,
+            },
+            ResultPath: temp,
+            Next: ASLGraph.DeferNext,
+            output: {
+              jsonPath: `${temp}.out`,
+            },
+          };
+        }
+      );
+    },
+  });
+
+  /**
+   * Use the States.ArrayRange intrinsic function to create a new array containing a specific range of elements.
+   * The new array can contain up to 1000 elements.
+   *
+   * @param start - first element in the new array
+   * @param end - final element of the new array (inclusive)
+   * @param
+   */
+  export const range = makeStepFunctionIntegration<
+    "SFN.Range",
```

Comment:
Should this be States.Range? To match the intrinsic name?

Reply:
They can.

Reply:
Though I am not using the full names of the intrinsics. (ArrayRange => Range) because "array" is covered by the type.